test/plugin/base_test.rb

```diff
@@ -2144,6 +2156,22 @@ def verify_subfields_when_not_hash(payload_key)
     end
   end
 
+  def verify_subfields_when_nil(payload_key)
+    destination_key = log_entry_subfields_params[payload_key][0]
+    @logs_sent = []
+    setup_gce_metadata_stubs
+    setup_logging_stubs do
+      d = create_driver
+      d.emit(payload_key => nil)
+      d.run
+    end
+
+    verify_log_entries(1, COMPUTE_PARAMS, 'jsonPayload') do |entry|
+      fields = get_fields(entry['jsonPayload'])
+      assert_nil fields[payload_key], entry[destination_key]
```

Comment:
I think you want to check that `fields` doesn't have the key anymore, to distinguish from the case where it's left there as a `nil`. You also want to check that the field is not present at the `destination_key`. Plus, there's no need to print one part of the `entry` — just dump the whole thing. So something like:
```ruby
      assert not fields.key?(payload_key), entry
      assert_nil entry[destination_key], entry
```

Reply:
+1

Reply:
Checked that the keys didn't exist in either hash instead, and used assert_false.

Reply:
FWIW, for the `entry` there's no difference between the key not existing and the key being set to `nil` — that is only user-visible in the `jsonPayload`. So I'd change the second `assert_false` to `assert_nil`, to be consistent with the rest of the tests (e.g., line 2155 above).

Reply:
Changed all asserts for absent LogEntry fields to check for a missing key as opposed to a nil value, since ruby protos will only have keys if they've been explicitly set.